go_v2/main.go

```diff
@@ -0,0 +1,84 @@
+/*
```

Comment:
It would be nice to switch it all to MIT, but you still have to have *something* in each file to claim copyright and refer to the license. The MIT is short so I took the path of least resistance. Would you like a PR where I either add the license to every file, or change the top level license and add an appropriate reference to each file?